core/api/serializers/cp_report.py

```diff
@@ -115,36 +119,18 @@ def to_representation(self, instance):
         return CPReportSerializer(instance).data
 
     # waiting for the decision on the validation rules
-    # def validate(self, attrs):
-    #     if attrs["year"] > IMPORT_DB_MAX_YEAR:
-    #         if not all(
-    #             [
-    #                 attrs.get("section_b"),
-    #                 attrs.get("section_d"),
-    #                 attrs.get("section_e"),
-    #                 attrs.get("section_f"),
-    #             ]
-    #         ):
-    #             raise serializers.ValidationError(
-    #                 f"Sections B, D, E and F are required for years after {IMPORT_DB_MAX_YEAR}"
-    #             )
-    #         if attrs["section_f"].get("remarks") is None:
-    #             raise serializers.ValidationError(
-    #                 f"Remarks are required for years after {IMPORT_DB_MAX_YEAR}"
-    #             )
-    #     else:
-    #         if not all(
-    #             [
-    #                 attrs.get("adm_b"),
-    #                 attrs.get("adm_c"),
-    #                 attrs.get("adm_d"),
-    #             ]
-    #         ):
-    #             raise serializers.ValidationError(
-    #                 f"Adm B, C and D are required for years before {IMPORT_DB_MAX_YEAR}"
-    #             )
-
-    #     return super().validate(attrs)
+    def validate(self, attrs):
+        if attrs.get("year") < VALIDATION_MIN_YEAR:
+            return super().validate(attrs)
+
+        validate_section_a(attrs.get("section_a"))
```

Comment:
I would call a single `validate_cp_report` from `../validations/cp_reports_validations.py` and let that handle the complexity. IIRC there might be some cross-section validation, also taking into account other non-section attrs.